rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java

```diff
@@ -157,7 +158,7 @@
   private static final int DEFAULT_REMOTE_FETCH_PROCESS_INTERVAL = 10000; // ms
   private static final int DEFAULT_REMOTE_FETCH_THREADS = 10;
   private static final boolean DEFAULT_RUBIX_ON_MASTER = false;
-  private static final int DEFAULT_SERVER_MAX_THREADS = Integer.MAX_VALUE;
+  private static final int DEFAULT_SERVER_MAX_THREADS = Runtime.getRuntime().availableProcessors() * 16;
```

Comment:
This is too low. With connection pooling, each connection has its own thread and in case of idle periods it won't consume resources. Lowering it to such a low value is going to disable connection pooling in a way. We should keep it higher, imo few thousands.

Reply:
Increased to 4k